prow/git/git.go

```diff
@@ -138,7 +137,17 @@ func (c *Client) Clone(repo string) (*Repo, error) {
 	base := c.base
 	user, pass := c.getCredentials()
 	if user != "" && pass != "" {
-		base = fmt.Sprintf("https://%s:%s@%s", user, pass, github)
+		u, err := url.Parse(c.base)
+		if err != nil {
+			return nil, err
+		}
+
+		nu := &url.URL{
```

Comment:
Could we use more descriptive variable names than `nu` please?

Reply:
Can't we just reuse `u`?

Reply:
Yes and Yes... please see new commit.